components/eventing-controller/controllers/subscriptionv2/jetstream/reconciler_internal_unit_test.go

```diff
@@ -0,0 +1,641 @@
+			},
+			wantStatus: false,
+		},
+		{
+			name: "A subscription with the same eventTypes and new jsTypes must return true",
+			givenSub: controllertesting.NewSubscription(subscriptionName, namespaceName,
+				controllertesting.WithStatus(true),
+				controllertesting.WithEventType(controllertesting.OrderCreatedUncleanEvent),
+				controllertesting.WithStatusTypes(eventTypes),
+			),
+			wantSubStatus: eventingv1alpha2.SubscriptionStatus{
+				Ready:   true,
+				Types:   eventTypes,
+				Backend: backendStatus,
+			},
+			wantStatus: true,
+		},
+		{
+			name: "A subscription with changed eventTypes and the same jsTypes must return true",
+			givenSub: controllertesting.NewSubscription(subscriptionName, namespaceName,
+				controllertesting.WithStatus(true),
+				controllertesting.WithEventType(controllertesting.OrderCreatedUncleanEvent),
+				controllertesting.WithStatusJSBackendTypes(jsTypes),
+			),
+			wantSubStatus: eventingv1alpha2.SubscriptionStatus{
+				Ready:   true,
+				Types:   eventTypes,
+				Backend: backendStatus,
+			},
+			wantStatus: true,
+		},
+	}
+	for _, tC := range testCases {
+		testCase := tC
+		t.Run(testCase.name, func(t *testing.T) {
+			// given
+			sub := testCase.givenSub
+
+			// when
+			gotStatus, err := r.syncInitialStatus(sub)
+			require.NoError(t, err)
+
+			// then
+			require.Equal(t, testCase.wantSubStatus.Types, sub.Status.Types)
+			require.Equal(t, testCase.wantSubStatus.Backend, sub.Status.Backend)
+			require.Equal(t, testCase.wantSubStatus.Ready, sub.Status.Ready)
+			require.Equal(t, testCase.wantStatus, gotStatus)
+		})
+	}
+}
+
+func Test_addFinalizerToSubscription(t *testing.T) {
+	// given
+	sub := controllertesting.NewSubscription(subscriptionName, namespaceName)
+	fakeSub := controllertesting.NewSubscription("fake", namespaceName)
+	testEnvironment := setupTestEnvironment(t, sub)
+	r := testEnvironment.Reconciler
+	fetchedSub, err := fetchTestSubscription(testEnvironment.Context, r)
+	require.NoError(t, err)
+	ensureFinalizerMatch(t, &fetchedSub, []string{})
+
+	testCases := []struct {
+		name      string
+		givenSub  *eventingv1alpha2.Subscription
+		wantError bool
+	}{
+		{
+			name:      "A new Subscription must be updated with cleanEventTypes and backend jstypes and return true",
+			givenSub:  sub,
+			wantError: false,
+		},
+		{
+			name:      "A new Subscription must be updated with cleanEventTypes and backend jstypes and return true",
+			givenSub:  fakeSub,
+			wantError: true,
+		},
+	}
+
+	for _, tC := range testCases {
+		testCase := tC
+		t.Run(testCase.name, func(t *testing.T) {
+			// when
+			err = r.addFinalizerToSubscription(testCase.givenSub, r.namedLogger())
+
+			// then
+			if testCase.wantError {
+				require.Error(t, err)
+			} else {
+				require.NoError(t, err)
+				fetchedSub, err = fetchTestSubscription(testEnvironment.Context, r)
+				require.NoError(t, err)
+				ensureFinalizerMatch(t, &fetchedSub, []string{eventingv1alpha2.Finalizer})
+			}
```

Comment:
(nit)
you case set the wantErr in the testStruct to the error type, have it nil in the first table test and `errFailedToDeleteSub` in the second and then adjust the test like this:
```suggestion
			if testCase.wantError != nil {
				require.ErrorIs(t, err, tC.wantError)
			} else {
				fetchedSub, err = fetchTestSubscription(testEnvironment.Context, r)
				require.NoError(t, err)
				ensureFinalizerMatch(t, &fetchedSub, []string{eventingv1alpha2.Finalizer})
			}```